File: II sem/Programowanie Obiektowe/Projekty/projekt2/po3/src/BarszczSosnowskiego.java
public class BarszczSosnowskiego extends Roslina {
    public BarszczSosnowskiego(String ikonka, int x_pos, int y_pos, int sila, int inicjatywa, Swiat swiat)
    {
        super(ikonka, x_pos, y_pos, sila, inicjatywa, swiat);
    }

    public BarszczSosnowskiego(Organizm organizm)
    {
        super(organizm);
    }

    @Override
    void kolizja(Organizm organizm)
    {

    }

    @Override
    public void moc(Organizm organizm)
    {
        organizm.zabijanie();
    }

    @Override
    Organizm kopiuj(Organizm organizm)
    {
        return new BarszczSosnowskiego(organizm);
    }
}

File: II sem/Programowanie Obiektowe/Projekty/projekt1/POproject1/BarszczSosnowskiego.cpp
#include "BarszczSosnowskiego.h"
#include "Zwierze.h"
#include "Swiat.h"


BarszczSosnowskiego::BarszczSosnowskiego(int x, int y) : Roslina(EMOJI ? u8"BS" : "BS", 10, x, y)
{

}

void BarszczSosnowskiego::akcja()
{
	for (int i = 0; i < 4; i++)
	{
		auto obok = swiat->getOrganizm({ getX() + poruszanie[i][0],getY() + poruszanie[i][1] });
		
		if (obok != nullptr)
		{
			bool czyZwierze = dynamic_cast<Zwierze*>(obok);

			if (czyZwierze)
			{
				swiat->dodajLog(this, " Otrucie:(");
				obok->zabijanie();
			}
		}
	}
}

void BarszczSosnowskiego::kolizja(Organizm* inny)
{
	string nazwa = typeid(*inny).name();
	swiat->dodajLog(this, " Zatrul " + nazwa.substr(6));
	inny->zabijanie();
	zabijanie();
}

Organizm* BarszczSosnowskiego::potomek()
{
	return new BarszczSosnowskiego(*this);
}